src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/ArrayHelpers.cs

```diff
@@ -60,16 +58,45 @@ public static unsafe Array NewObjArray(IntPtr pEEType, int nDimensions, int* pDi
                     }
```

Comment:
The `(eeType.IsSzArray)` from above should be moved to the other helper as well.

Reply:
Actually, the condition in the JIT does not look right either. It will make us use the special helper for the jagged arrays - we do not want that.

Reply:
I do not see a method on JIT/EE interface that can be used to distinguish 2-dimensional jagged array vs. 1-dimensional md-array with two args. We may need a new method on the JIT/EE interface for this.

Reply:
Can the jagged arrays be constructed with regular C#? We could just call into the old helper to handle the rest of cases. It's really just about splitting off the edge cases from the mainstream thing.

Reply:
Good point. It makes sense to have a helper that handles just what is used by C# - true MD arrays of rank 2 or more, and second helper for everything else.